Applications/Terminal/Terminal.cpp

```diff
@@ -254,7 +303,13 @@ void Terminal::escape$r(const ParamVector& params)
         top = params[0];
     if (params.size() >= 2)
         bottom = params[1];
-    dbgprintf("FIXME: escape$r: Set scrolling region: %u-%u\n", top, bottom);
+    if ((bottom - top) < 2 || bottom > m_rows) {
```

Comment:
Why 2? I'd guess that a single line region would be valid. Might deserve a comment?

Missing a check for top being too small?

Reply:
So according to the reference material I have, scroll regions are not allowed to have less than two rows. Your guess is as good as mine concerning why...